reascripts/ReaSpeech/source/include/globals.lua

```diff
@@ -12,12 +12,20 @@ for name, func in pairs(reaper) do
 end
 
 -- For debugging
-function dump(o)
+function dump(o, seen_tables)
+  seen_tables = seen_tables or {}
   if type(o) ~= "table" then return tostring(o) end
+
+  if seen_tables[tostring(o)] then
+    return "{ circular reference detected }"
+  end
+
+  seen_tables[tostring(o)] = true
+
   local result = {"{"}
   for k, v in pairs(o) do
-    k = type(k) == "string" and k or "[" .. dump(k) .. "]"
-    v = type(v) == "string" and '"' .. v .. '"' or dump(v)
+    k = type(k) == "string" and k or "[" .. dump(k, seen_tables) .. "]"
+    v = type(v) == "string" and '"' .. v .. '"' or dump(v, seen_tables)
```

Comment:
this was quick'n'dirty. i could see adding some context to the "circular reference" message, perhaps with the table's `tostring()` output.

Reply:
> Works for me. I'm curious how this came up?

iirc it was while providing parent references to child objects - so i was getting like
```
app.plugins[n].app.💥
```